src/main/java/redis/clients/jedis/JedisClusterCommand.java

```diff
@@ -98,15 +99,17 @@ private T runWithRetries(final int slot) {
       } catch (JedisNoReachableClusterNodeException e) {
         throw e;
       } catch (JedisConnectionException e) {
+        lastException = e;
         connectionSupplier = handleConnectionProblem(slot, currentAttempt);
       } catch (JedisRedirectionException e) {
+        lastException = e;
         redirectionSupplier = handleRedirection(connection, e);
       } finally {
         releaseConnection(connection);
       }
     }
 
-    throw new JedisClusterMaxAttemptsException("No more cluster attempts left.");
+    throw new JedisClusterMaxAttemptsException("No more cluster attempts left.", lastException);
```

Comment:
Just adding another idea. What if we add the `lastException` as suppressed, not as cause?
@mina-asham WDYT?

Reply:
Ah, that's an even better idea as it allows us to have the full list of exceptions, let's do that!

Reply:
NO. Let's start with one and see if uses want more.

Check resolves and closes of #2233. You'll get an idea.

Reply:
It doesn't cost much to add all of them and it's more informative (but I might be biases by my own opinion as a user here), in anyway I don't mind either approach for now, it's a two way door in either case :)